### ISSUE TITLE
when i use custom loss and gpu, fit kernel died

### ISSUE THREAD

**Issue by YooSungHyun** [keras-team/keras#14366]
my custom loss function looks like this,

```
import tensorflow as tf
from keras import backend as K
from tensorflow.keras.losses import Loss

class MASE(Loss): #inherit parent class
    #class attribute
    seasonality = 1
  
    #initialize instance attributes
    def __init__(self, seasonality):
        super().__init__()
        self.seasonality = seasonality

    def _naive_forecasting(self, actual, seasonality: int = 1):
        return actual[:-seasonality]
    
    def _error(self, actual, predicted):
        return actual - predicted
    
    def _mae(self, actual, predicted):
        return K.mean(K.abs(self._error(actual, predicted)))

    #compute loss
    def call(self, y_true, y_pred):
        return self._mae(y_true, y_pred) / self._mae(y_true[self.seasonality:], self._naive_forecasting(y_true, self.seasonality))
```

and use like this,

```
model.compile(loss=MASE(1), optimizer=Adam(lr=0.001))
...
model.fit(x_concat_data[train], y_concat_data[train], batch_size=batch_size, epochs=epoch, verbose=2, shuffle=True)
...
model.evaluate(x_concat_data[validation], y_concat_data[validation], batch_size=batch_size, callbacks=[early_stopping])
```

I use rtx 3090 and want to train with gpu

cpu training is good!
BUT! when i use gpu, python kernel is dead!
kernel output like this

2021-01-14 09:33:55.296929: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:127] None of the MLIR optimization passes are enabled (registered 2)
Epoch 1/1000
d:/>

what is my problem?
im use tf-nightly-gpu 2.5.0 and python 3.8

**Comment by YooSungHyun** [comment]
I uninstalled tf-nightly version, and installing tensorflow-gpu 2.4.0.
that version work good

issue happend only tf-nightly version. maybe i think that is nightly version`s bug... but why happened like that.

i will use tensorflow-gpu 2.4.0. so, this issue is complete to me.
